source/state_representation/src/trajectory/CartesianTrajectory.cpp

```diff
@@ -0,0 +1,210 @@
+#include "state_representation/trajectory/CartesianTrajectory.hpp"
+#include "state_representation/space/cartesian/CartesianState.hpp"
+
+#include "state_representation/exceptions/EmptyStateException.hpp"
+#include "state_representation/exceptions/IncompatibleReferenceFramesException.hpp"
+
+namespace state_representation {
+CartesianTrajectory::CartesianTrajectory() {
+  this->set_type(StateType::CARTESIAN_TRAJECTORY);
+}
+
+CartesianTrajectory::CartesianTrajectory(const std::string& name, const std::string& reference_frame)
+    : TrajectoryBase<CartesianTrajectoryPoint>(name), reference_frame_(reference_frame) {
+  this->set_type(StateType::CARTESIAN_TRAJECTORY);
+}
+
+CartesianTrajectory::CartesianTrajectory(
+    const std::string& name, const CartesianState& point, const std::chrono::nanoseconds& duration
+)
+    : TrajectoryBase<CartesianTrajectoryPoint>(name) {
+  this->set_type(StateType::CARTESIAN_TRAJECTORY);
+  if (point.is_empty()) {
+    throw exceptions::EmptyStateException("The Cartesian state provided is empty");
+  }
+  this->reference_frame_ = point.get_reference_frame();
+  this->add_point(point, duration);
```

Comment:
```suggestion
    : CartesianTrajectory(name, point.get_reference_frame()) {
  this->add_point(point, duration);
```

Reply:
Here we actually need to do some checks so that we retrieve the reference frame (or joint names for JointTrajectory)

I am indeed duplicating a bit, but since `add_point` (and related functions) are not allowed to change this variables, we need to duplicate checks in the constructor I think